### PR TITLE
V3

### PR DESCRIPTION
**Pull request by mattcgenui**
**Summary:**

Having numerous issues with build on new Elastic Beanstalk environment and previously with updating to Angular 13+ / Typescript. We decided to start from a fresh Angular 13 install and add back components one by one.

**General:**

- Updated all packages
- Moved shell scripts to /scripts
- Moved from tslint to eslint
- Cleaned config files (old)

**Server:**

- Updated all dependencies
- Splitted API routes to separate files under server/components
- Removed unused routes
- Move Schemas under server/models
- Updated code for Typescript 4.6.4
- Added node-cache (https://www.npmjs.com/package/node-cache)
- Added CSP rules

**App:**

- Updated folder structure
- Updated code for Typescript 4.6.4
- Removed all unused components/services/code
- Updated/optimized all the charts/plots
- Implemented New design

**CI:**

Coming soon

**TODO**
- Karma tests will be added back soon
- Finish to update readme

### CODE REVIEW

**Review by zaro0508 — CHANGES_REQUESTED** [review]
from my quick skim..

why have these files been deleted? 
`.gitattributes, .github/CONTRIBUTING.md, .gitignore, .nvmrc`


why did you add the `.vscode` folder?  Aren't those each developer's IDE specific settings?

**Comment by mattcgenui** [comment]
Thank you @zaro0508, I added `.gitattributes, .github/CONTRIBUTING.md, .nvmrc` back. I will make sure we have all the necessary config files during the pipeline work.

I believe the .vscode was already there but my IDE added/updated files. eg.g `extensions.json` should suggests plugins to install while working on the project

**Review by zaro0508 — APPROVED** [review]
let it ride!

File: package.json [hunk 1]
```diff
@@ -76,16 +62,15 @@
...
+    "server:createindexes": "mongo --host localhost ./src/server/scripts/createIndexes.js",
     "start": "node --max_old_space_size=2000  ./dist/server.js",
-    "start:mongo": "sh start-mongo.sh '/var/lib/mongodb/'",
-    "start:mongo:windows": "sh start-mongo.sh 'C:\\data\\db'",
+    "start:mongo": "sh scripts/start-mongo.sh '/var/lib/mongodb/'",
+    "start:mongo:windows": "sh scripts/start-mongo.sh 'C:\\data\\db'",
     "start:prod": "npm run watch:server:prod",
     "start:hmr": "npm run server:dev:hmr",
-    "test": "npm run lint && karma start",
+    "test": "npm run lint",
```
Comment (jay-hodgson) [inline package.json:71]: Does this mean that after this change there are no automated tests being run?

Comment (mattcgenui) [reply package.json:71]: Hi @jay-hodgson, I need to add the tests back (this should be done by Friday). I temporary removed karma test so we can test the Travis pipeline / EB deployment

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**